arviz/stats/stats.py

```diff
@@ -239,16 +239,18 @@ def compare(
     p_ic = f"p_{ic}"
     ic_i = f"{ic}_i"
 
-    ics = pd.DataFrame()
+    ics = {}
     names = []
     for name, dataset in dataset_dict.items():
         names.append(name)
         try:
             # Here is where the IC function is actually computed -- the rest of this
             # function is argument processing and return value formatting
-            ics = ics.append([ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
+            ics[name] = ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)
         except Exception as e:
             raise e.__class__(f"Encountered error trying to compute {ic} from model {name}.") from e
+
+    ics = pd.concat(ics, axis=1).T
     ics.index = names
```

Comment:
I think this line might be unnecessary now?

Reply:
I am replacing the initialisation of `ics` from `pd.DataFrame()` to `dict`. So this line makes it a DataFrame.

Reply:
sorry I meant the line after (`ics.index = names`)

Reply:
Right, yes. I'll delete it.